airbyte-config/config-models/src/main/resources/type_exploration/airbyte.proto

```diff
@@ -0,0 +1,76 @@
+import "google/protobuf/timestamp.proto";
+
+// =============================== DEFINITIONS ================================
+// types supported by airbyte.
+enum FieldType {
+  STRING = 0;
+  INT = 1;
+  LONG = 2;
+  BOOLEAN = 3;
+  OBJECT = 4;
+  LIST = 5;
+  DATE = 6;
+  TIMESTAMP_WITH_TIMEZONE = 7;
+  TIMESTAMP_WITHOUT_TIMEZONE = 8;
+  TIME_WITH_TIMEZONE = 9;
+  TIME_WITHOUT_TIMEZONE = 10;
+}
+
+// custom struct for each type
+message AirbyteStringField {
+  required string value = 1;
+}
+
+message AirbyteIntField {
+  required int32 value = 1;
+}
+
+message AirbyteTimestampWithTimezone {
+  required google.protobuf.Timestamp timestamp = 1;
+}
+
+message AirbyteTimestampWithoutTimezone {
+  required google.protobuf.Timestamp timestamp = 1;
+}
+
+message AirbyteTimeWithTimezone {
+  required google.protobuf.Timestamp timestamp = 1;
+}
+
+message AirbyteTimeWithoutTimezone {
+  required google.protobuf.Timestamp timestamp = 1;
+}
+
+// ====================== EXAMPLE SCHEMA IN PROTOBUF ===========================
+
+// example schema we want to be able to model
+// users
+// id - int
+// name - string
+// created_at - timestamp with timezone
+// tags - list strings
+// friends - list object { id, name }
+// address - object { address_num, street, city, zip }
+
+message address {
+  optional AirbyteIntField address_num = 1;
+  optional AirbyteStringField street = 2;
+  optional AirbyteStringField city = 3;
+  optional AirbyteIntField zip = 4;
```

Comment:
not that it's a huge deal, but massachusetts is "02148" so zip codes are usually strings.